BenchmarkTests/Runner/Scenarios/RUM/RUMAuto/UI/Locations/LocationsViewController.swift

```diff
@@ -0,0 +1,166 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import SwiftUI
+import UIKit
+
+class LocationsViewController: UIViewController {
+    @IBOutlet var collectionView: UICollectionView!
+    @IBOutlet var activityIndicator: UIActivityIndicatorView!
+
+    private var locations: [Location] = []
+    private var nextPageURL: String?
+    private var isLoading = false
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupCollectionView()
+        setupNavigationBar()
+        fetchLocations()
+    }
+
+    private func setupNavigationBar() {
+        navigationController?.navigationBar.prefersLargeTitles = true
+        navigationItem.largeTitleDisplayMode = .always
+    }
+
+    private func setupCollectionView() {
+        if let flowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            flowLayout.sectionInset = UIEdgeInsets(top: 16, left: 16, bottom: 16, right: 16)
+
+            let width = UIScreen.main.bounds.width - 32
+            flowLayout.itemSize = CGSize(width: width, height: 120)
+        }
+    }
+
+    /// Fetches locations from the API
+    private func fetchLocations() {
+        guard !isLoading else { return }
+
+        isLoading = true
+        activityIndicator.startAnimating()
+
+        Task {
+            do {
+                let result = try await RickMortyService.shared.fetchLocations(nextPageURL: nextPageURL)
+
+                DispatchQueue.main.async { [weak self] in
```

Comment:
/suggestion To fully follow the Concurrency pattern, lets use [`@MainActor`](https://developer.apple.com/documentation/swift/mainactor) methods to update the UI.